cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc

```diff
@@ -0,0 +1,334 @@
+      "Action": [
+        "secretsmanager:GetSecretValue",
+        "secretsmanager:DescribeSecret"
+      ],
+      "Resource": ["$SECRET_ARN"]
+      }]
+}
+EOF
+----
+
+. Create an IAM Access Policy:
++
+[source,terminal]
+----
+$ POLICY_ARN=$(aws --region "$REGION" --query Policy.Arn \
+   --output text iam create-policy \
+   --policy-name openshift-access-to-mysecret-policy \
+   --policy-document file://policy.json)
+$ echo $POLICY_ARN
+----
+
+. Create an IAM Role trust policy document:
++
+[NOTE]
+====
+The trust policy is locked down to the default service account of a namespace you will create later.
+====
++
+[source,terminal]
+----
+$ cat <<EOF > trust-policy.json
+{
+   "Version": "2012-10-17",
+   "Statement": [
+   {
+   "Effect": "Allow",
+   "Condition": {
+     "StringEquals" : {
+       "${OIDC_ENDPOINT}:sub": ["system:serviceaccount:my-application:default"]
+      }
+    },
+    "Principal": {
+       "Federated": "arn:aws:iam::$AWS_ACCOUNT_ID:oidc-provider/${OIDC_ENDPOINT}"
+    },
+    "Action": "sts:AssumeRoleWithWebIdentity"
+    }
+    ]
+}
+EOF
+----
+
+. Create an IAM role:
++
+[source,terminal]
+----
+$ ROLE_ARN=$(aws iam create-role --role-name openshift-access-to-mysecret \
+    --assume-role-policy-document file://trust-policy.json \
+    --query Role.Arn --output text)
+$ echo $ROLE_ARN
+----
+
+. Attach the role to the policy:
++
+[source,terminal]
+----
+$ aws iam attach-role-policy --role-name openshift-access-to-mysecret \
+    --policy-arn $POLICY_ARN
+----
+
+[id="rosa-mobb-aws-secret-manager-creating-application"]
+== Create an Application to use this secret
+
+. Create an OpenShift project:
++
+[source,terminal]
+----
+$ oc new-project my-application
+----
+
+. Annotate the default service account to use the STS Role:
++
+[source,terminal]
+----
+$ oc annotate -n my-application serviceaccount default \
+    eks.amazonaws.com/role-arn=$ROLE_ARN
+----
+
+. Create a secret provider class to access our secret:
++
+[source,terminal]
+----
+$ cat << EOF | oc apply -f -
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-application-aws-secrets
+spec:
+  provider: aws
+  parameters:
+    objects: |
+      - objectName: "MySecret"
+        objectType: "secretsmanager"
+EOF
+----
+
+. Create a Deployment using our secret:
++
+[source,terminal]
+----
+$ cat << EOF | oc apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-application
+  labels:
+    app: my-application
+spec:
+  volumes:
+  - name: secrets-store-inline
+    csi:
+      driver: secrets-store.csi.k8s.io
+      readOnly: true
+      volumeAttributes:
+        secretProviderClass: "my-application-aws-secrets"
+  containers:
+  - name: my-application-deployment
+    image: k8s.gcr.io/e2e-test-images/busybox:1.29
+    command:
+      - "/bin/sleep"
+      - "10000"
+    volumeMounts:
+    - name: secrets-store-inline
+      mountPath: "/mnt/secrets-store"
+      readOnly: true
+EOF
```

Comment:
This fails due to a pod security admission controller. We need to update this deployment to address that (and we need to update all of our secrets store CSI docs to do that). Will circle back on this one.

Reply:
This is fixed with the changes above.

Reply:
@michaelryanmcneill So to clarify, I do not need to change anything from the highlighted lines 249 to 279?

Reply:
Correct!